File: 0198-house-robber/0198-house-robber.go
func rob(nums []int) int {
    notRob, rob := 0, nums[0]
    for i:=1; i<len(nums); i++ {
        notRobCurr := max(notRob, rob)
        rob = notRob + nums[i]
        notRob = notRobCurr
    }

    return max(rob, notRob)
}

File: 0198-house-robber/0198-house-robber.cs
public class Solution {
    public int Rob(int[] nums) {
        int notRob = 0, rob = nums[0];
        for(var i=1; i<nums.Length; i++){
            var notRobCurr = Math.Max(notRob, rob);
            rob = notRob + nums[i];
            notRob = notRobCurr;
        }

        return Math.Max(rob, notRob);
    }
}